src/tqec/circuit/detectors/match.py

```diff
@@ -0,0 +1,489 @@
+
+    Args:
+        flows: a list of fragment flow to search detectors in.
+        qubit_coordinates: a mapping from qubit indices to coordinates. Used to annotate
+            the matched detectors with the coordinates from the qubits involved in the
+            measurement forming the detector.
+
+    Returns:
+        the list of all the detectors found. These detectors are only valid if inserted
+        at the end of the fragment represented by the last provided flow in the input
+        parameter `flows`. If the last provided flow is an instance of
+        :class:`FragmentLoopFlows` then the returned detectors should be inserted at
+        the end of the loop body.
+    """
+    detectors: list[list[MatchedDetector]] = [
+        match_detectors_within_fragment(flow, qubit_coordinates) for flow in flows
+    ]
+    for i in range(1, len(flows)):
+        detectors[i].extend(
+            match_boundary_stabilizers(flows[i - 1], flows[i], qubit_coordinates)
+        )
+
+    return detectors
+
+
+def match_detectors_within_fragment(
+    flows: FragmentFlows | FragmentLoopFlows,
+    qubit_coordinates: dict[int, tuple[float, ...]],
+) -> list[MatchedDetector]:
+    """Match all the detectors that can be resolved without considering neighbouring
+    fragments.
+
+    Some detectors only include measurements from one fragment. This is for example the
+    case during the initialisation of any code: some measurements from the first round
+    have a pre-defined value that should be constant in the absence of errors.
+
+    Note that this function ignores "trivial" flows to avoid trivial detectors to be
+    matched on some particular cases (e.g., 1-round repetition code).
+
+    Args:
+        flows: pre-computed flows for the fragment of interest. If any detector
+            is found, the involved flow(s) will be removed from the provided instance.
+            This means that **this parameter may be mutated inline**.
+        qubit_coordinates: a mapping from qubit indices to coordinates. Used to annotate
+            the matched detectors with the coordinates from the qubits involved in the
+            measurement forming the detector.
+
+    Returns:
+        the list of all the detectors found.
+    """
+    matched_detectors: list[MatchedDetector] = []
+
+    matched_detectors.extend(
+        _match_non_propagating_non_trivial_flows_inline(
+            flows.creation, qubit_coordinates
+        )
+    )
+    matched_detectors.extend(
+        _match_non_propagating_non_trivial_flows_inline(
+            flows.destruction, qubit_coordinates
+        )
+    )
+    return matched_detectors
+
+
+def _match_non_propagating_non_trivial_flows_inline(
+    boundary_stabilizers: list[BoundaryStabilizer],
+    qubit_coordinates: dict[int, tuple[float, ...]],
+) -> list[MatchedDetector]:
+    """Match all the detectors that can be trivially resolved and remove the
+    matched boundary stabilizers from the provided list.
```

Comment:
Then the name is fine, I was confused because the docstrings say `detectors that can be trivially resolved` which is not the same, or is it?

Reply:
Yes, there are two usage of the word "trivial" that do not mean the same thing here. Also I think that the concept of "trivial flows" will eventually disappear from the code base.
The `detectors that can be trivially resolved` use "trivially" because the implemented matching is quite trivial: just look if the collapsing operations are terminating the flow propagation, and if so this is a detector. I will change that phrasing.

Reply:
I added so precision in the docstring.